### PR TITLE
fix(test-ssr): do not transform JSX in preparation stage

### PR DESCRIPTION
**Pull request by layershifter**
## New Behavior

Follow up for #30123.

- Updates the part of SSR tests that get stories exports to avoid JSX transforms, so there will be no pragma-related errors
- Fixes the story introduces in #30123
- Adds error intercepting for HTML rendering (aka SSR part)

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/4d269ba336f4a4cd9d1b5ab06949baacc9dc104d'>4d269ba336f4a4cd9d1b5ab06949baacc9dc104d</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/Avatar-mount_base.html">649</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/Avatar-mount.html">641</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/Button-mount_base.html">294</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/Button-mount.html">295</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/Field-mount_base.html">1083</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/Field-mount.html">1092</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/FluentProvider-mount_base.html">679</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/FluentProvider-mount.html">680</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/FluentProviderWithTheme-mount_base.html">82</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/FluentProviderWithTheme-mount.html">80</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">58</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">58</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">78</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">72</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/MakeStyles-mount_base.html">862</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/MakeStyles-mount.html">867</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/Persona-mount_base.html">1716</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/Persona-mount.html">1735</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/SpinButton-mount_base.html">1355</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30131/perf-test-react-components/SpinButton-mount.html">1419</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by bsunderhus — APPROVED** [review]

File: scripts/test-ssr/src/utils/getExportFromFile.ts [hunk 1]
```diff
@@ -91,7 +91,7 @@ export async function getExportFromFile(distDir: string, filename: string): Prom
     },
   };
 
-  await Babel.transformAsync(sourceCode, {
+  await Babel.transformSync(sourceCode, {
```
Comment (Hotell) [inline scripts/test-ssr/src/utils/getExportFromFile.ts:94]: why this change ? if we need sync isnt that await redundant ?

File: scripts/test-ssr/src/commands/main.ts [hunk 2]
```diff
@@ -27,6 +28,36 @@ type MainParams = {
...
+    // Ignoring 'aria-description' warning from React 17 as it's a valid prop
+    // https://github.com/facebook/react/issues/21035
+    if (containsAriaDescriptionWarning(args[0] + ' ' + args[1]) && React.version.startsWith('17')) {
+      return;
+    }
+
+    console.log(chalk.bgRed.whiteBright('❌  A console.error() was thrown during rendering HTML:'));
+    originalConsoleError(...args);
+
+    throw new Error('During rendering a console.error() was thrown');
```
Comment (Hotell) [inline scripts/test-ssr/src/commands/main.ts:45]: is this necessary ? as it's kinda dupe of line 42 ?

File: scripts/test-ssr/src/commands/main.ts [hunk 3]
```diff
@@ -27,6 +28,36 @@ type MainParams = {
...
+  };
+
+  console.warn = (...args: unknown[]) => {
```
Comment (Hotell) [inline scripts/test-ssr/src/commands/main.ts:48]: why do we need to hijack console.warn when there is no extra logic needed ?

File: scripts/test-ssr/src/utils/getExportFromFile.ts (hunk 1 above)
Comment (layershifter) [reply scripts/test-ssr/src/utils/getExportFromFile.ts]: Good catch, I reverted it

File: scripts/test-ssr/src/commands/main.ts (hunk 3 above)
Comment (layershifter) [reply scripts/test-ssr/src/commands/main.ts:48]: To ensure that `console.warn()` will not be called in tests

File: scripts/test-ssr/src/commands/main.ts (hunk 2 above)
Comment (layershifter) [reply scripts/test-ssr/src/commands/main.ts:45]: image


To clarify:
- _45_ is needed to throw an actual exception and break the test
- _43_ is needed to log the original output as otherwise it's unclear what happens
- _42_ is needed to make _43_ in the output visible

Does it make sense?

Comment (Hotell) [reply scripts/test-ssr/src/commands/main.ts:45]: yeah I understand the logic, i was rather pointing to the message thrown on line 45. as its not being printed anywhere ?

also the message in your print screen is bit confusing , "error below contains relevant info" -> should be "above" ?

File: scripts/test-ssr/src/commands/main.ts (hunk 3 above)
Comment (Hotell) [reply scripts/test-ssr/src/commands/main.ts:48]: how come console.warn throws an error ? is it some internal reacts logging logic ?

or we wanna throw on console warnings ?

File: scripts/test-ssr/src/commands/main.ts (hunk 2 above)
Comment (layershifter) [reply scripts/test-ssr/src/commands/main.ts:45]: > as its not being printed anywhere ?

It's printed below, it didn't fit on the screenshot.

> also the message in your print screen is bit confusing , "error below contains relevant info" -> should be "above" ?

Kinda agree, however there is a stacktrace below

File: scripts/test-ssr/src/commands/main.ts (hunk 3 above)
Comment (layershifter) [reply scripts/test-ssr/src/commands/main.ts:48]: > or we wanna throw on console warnings ?

We want to throw on warnings, too. There should be no warnings in production mode in Node or in a browser.

Comment (Hotell) [reply scripts/test-ssr/src/commands/main.ts:48]: gotcha, can we document this intent as part of that function JSDoc ?

**Review by Hotell — APPROVED** [review]

File: scripts/test-ssr/src/commands/main.ts (hunk 3 above)
Comment (layershifter) [reply scripts/test-ssr/src/commands/main.ts:52]: Yup, done
